NBitcoin.Altcoins/Stratis.cs

```diff
@@ -0,0 +1,517 @@
+
+		public override string CryptoCode => "STRAT";
+
+		private Stratis()
+		{
+
+		}
+
+#pragma warning disable CS0618 // Type or member is obsolete
+		public class StratisConsensusFactory : ConsensusFactory
+		{
+			private StratisConsensusFactory()
+			{
+			}
+
+			public static StratisConsensusFactory Instance { get; } = new StratisConsensusFactory();
+
+			public override Block CreateBlock()
+			{
+				return new StratisBlock(new BlockHeader());
+			}
+
+			public override Transaction CreateTransaction()
+			{
+				return new StratisTransaction(this);
+			}
+		}
+
+		public class StratisBlockSignature : IBitcoinSerializable
+		{
+			protected bool Equals(StratisBlockSignature other)
+			{
+				return Equals(signature, other.signature);
+			}
+
+			public override bool Equals(object obj)
+			{
+				if (ReferenceEquals(null, obj)) return false;
+				if (ReferenceEquals(this, obj)) return true;
+				if (obj.GetType() != this.GetType()) return false;
+				return Equals((StratisBlockSignature)obj);
+			}
+
+			public override int GetHashCode()
+			{
+				return (signature?.GetHashCode() ?? 0);
+			}
+
+			public StratisBlockSignature()
+			{
+				this.signature = new byte[0];
+			}
+
+			private byte[] signature;
+
+			public byte[] Signature
+			{
+				get
+				{
+					return signature;
+				}
+				set
+				{
+					signature = value;
+				}
+			}
+
+			internal void SetNull()
+			{
+				signature = new byte[0];
+			}
+
+			public bool IsEmpty()
+			{
+				return !this.signature.Any();
+			}
+
+			public static bool operator ==(StratisBlockSignature a, StratisBlockSignature b)
+			{
+				if (System.Object.ReferenceEquals(a, b))
+					return true;
+
+				if (((object)a == null) || ((object)b == null))
+					return false;
+
+				return a.signature.SequenceEqual(b.signature);
+			}
+
+			public static bool operator !=(StratisBlockSignature a, StratisBlockSignature b)
+			{
+				return !(a == b);
+			}
+
+			#region IBitcoinSerializable Members
+
+			public void ReadWrite(BitcoinStream stream)
+			{
+				stream.ReadWriteAsVarString(ref signature);
+			}
+
+			#endregion
+
+			public override string ToString()
+			{
+				return Encoders.Hex.EncodeData(this.signature);
+			}
+		}
+
+		public class StratisBlock : Block
+		{
+			public StratisBlock(BlockHeader header) : base(header)
+			{
+
+			}
+			public override ConsensusFactory GetConsensusFactory()
+			{
+				return StratisConsensusFactory.Instance;
+			}
+
+			public override void ReadWrite(BitcoinStream stream)
+			{
+				base.ReadWrite(stream);
+				stream.ReadWrite(ref this.blockSignature);
+			}
+
+			public static bool BlockSignature = false;
+
+			// block signature - signed by one of the coin base txout[N]'s owner
+			private StratisBlockSignature blockSignature = new StratisBlockSignature();
+
+			public StratisBlockSignature BlockSignatur
+			{
+				get { return this.blockSignature; }
+				set { this.blockSignature = value; }
+			}
+		}
+
+		private class StratisWitness
```

Comment:
We currently don't support witness so I suggest not adding witness at all, you just need to serialize the trx as is and att the `stream.ReadWrite(ref nTimeTemp);`

Reply:
hmm well if you got it working already and its working fine then leave it as is.